### PR TITLE
Added GetLdapGroups func

### PR DESCRIPTION
**Pull request by tomercy**

### CODE REVIEW

**Review by mickeykcyolo — COMMENTED** [review]
Looks pretty good.
left some remarks along the code.

1. Please check it with open ldap and active directory.
2. add a test. The easiest way to check it would be to docker run openldap container:
"docker run --privileged -vvv -p 389:389 rroemhild/test-openldap".
follow my test of runQueries.

File: ldap-client.go [hunk 1]
```diff
@@ -202,3 +202,28 @@ func (lc *LDAPClient) GetGroupsOfUser(username string) ([]string, error) {
 	}
 	return groups, nil
 }
+
+// GetLdapGroups returns list of groups matching a name.
+func (lc *LDAPClient) GetLdapGroups(groupName string) ([]string, error) {
```
Comment (mickeykcyolo) [inline ldap-client.go:207]: I would change to getAllGroups as it is obvious that it's ldap.

File: ldap-client.go [hunk 2]
```diff
@@ -202,3 +202,28 @@ func (lc *LDAPClient) GetGroupsOfUser(username string) ([]string, error) {
...
+	err := lc.Connect()
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (mickeykcyolo) [inline ldap-client.go:212]: please add here the bind block:
// First bind with a read only user
	if lc.BindDN != "" && lc.BindPassword != "" {
		if err = lc.Conn.Bind(lc.BindDN, lc.BindPassword); err != nil {
			return
		}
	}

File: ldap-client.go [hunk 3]
```diff
@@ -202,3 +202,28 @@ func (lc *LDAPClient) GetGroupsOfUser(username string) ([]string, error) {
...
+	searchRequest := ldap.NewSearchRequest(
+		lc.Base,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		fmt.Sprintf("(&(objectCategory=Group)(distinguishedName=*%s*))", groupName),
+		[]string{"cn"}, // can it be something else than "cn"?
```
Comment (mickeykcyolo) [inline ldap-client.go:217]: i think you need the dn here

File: ldap-client.go [hunk 4]
```diff
@@ -202,3 +202,28 @@ func (lc *LDAPClient) GetGroupsOfUser(username string) ([]string, error) {
...
+		nil,
+	)
+	sr, err := lc.Conn.Search(searchRequest)
+	if err != nil {
+		return nil, err
+	}
+	var groups []string
+	for _, entry := range sr.Entries {
+		groups = append(groups, entry.GetAttributeValue("cn"))
```
Comment (mickeykcyolo) [inline ldap-client.go:226]: do you want to get the cn here or the dn?
cn are more user friendly.

**Review by mickeykcyolo — CHANGES_REQUESTED** [review]

File: ldap-client.go (hunk 1 above)
Comment (tomercy) [reply ldap-client.go:207]: Changed to GetAllGroupsByName

File: ldap-client.go (hunk 2 above)
Comment (tomercy) [reply ldap-client.go:212]: added

File: ldap-client.go (hunk 4 above)
Comment (tomercy) [reply ldap-client.go:226]: dn, ty

File: ldap-client.go (hunk 3 above)
Comment (tomercy) [reply ldap-client.go:217]: yes

**Review by mickeykcyolo — APPROVED** [review]

**Pull request merged**